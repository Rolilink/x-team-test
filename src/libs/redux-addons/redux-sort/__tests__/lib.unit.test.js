import withSorting, { sortingFactory, cases, selectors } from '../';

const { clearSort, setField, unsetField } = cases;
const { getFields, getField } = selectors();

const defaultState = {
  items: [],
  sort: {},
};

describe('cases', () => {
  describe('clearSort', () => {
    it('should set the sort subtree to an empty object', () => {
      const state = { ...defaultState };
      const newState = clearSort(state, {});

      expect(newState).toMatchSnapshot();
    });
  });


  describe('setField', () => {
    it('should add the specified record to the sort subtree', () => {
      const state = { ...defaultState, sort: { size: 'asc' } };
      const action = { payload: { sort: { name: 'asc' } } };
      const newState = setField(state, action);

      expect(newState).toMatchSnapshot();
    });

    it('should update the specified record from the sort subtree if it exist', () => {
      const state = { ...defaultState, sort: { size: 'asc', name: 'asc' } };
      const action = { payload: { sort: { name: 'desc' } } };
      const newState = setField(state, action);

      expect(newState).toMatchSnapshot();
    });
  });

  describe('unsetField', () => {
    it('should remove the specified record from the sort subtree', () => {
      const state = { ...defaultState, sort: { size: 'asc', name: 'asc' } };
      const action = { payload: { sort: { key: 'size' } } };
      const newState = unsetField(state, action);

      expect(newState).toMatchSnapshot();
    });
  });
});

describe('selectors', () => {
  describe('getFields', () => {
    it('should return all the sorts as an array', () => {
      const state = { ...defaultState, sort: { size: 'asc', name: 'asc' } };
      const result = getFields(state);

      expect(result).toMatchSnapshot();
    });
  });

  describe('getField', () => {
    it('should return the sort status of the key passed as param', () => {
      const state = { ...defaultState, sort: { size: 'asc', name: 'desc' } };
      const result = getField(state, { field: 'name' });

      expect(result).toMatchSnapshot();
    });
  });
});

describe('withSorting', () => {
  it('should enhance the state with sort state', () => {
    const state = { items: [] };
    const newState = withSorting(state);

    expect(newState).toMatchSnapshot();
  });

  it('should be able to initialize the sort state ', () => {
    const state = { items: [] };
    const initialSortingState = { name: 'asc' };
    const newState = withSorting(state, initialSortingState);

    expect(newState).toMatchSnapshot();
  });
});

describe('sortingFactory', () => {
  /* TODO: create passing properties scenario */
  it('should return default sort state', () => {
    const sort = sortingFactory();

    expect(sort).toMatchSnapshot();
  });
});
