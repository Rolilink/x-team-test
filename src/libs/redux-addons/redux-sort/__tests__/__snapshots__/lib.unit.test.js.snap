// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cases clearSort should set the sort subtree to an empty object 1`] = `
Object {
  "items": Array [],
  "sort": Object {},
}
`;

exports[`cases setField should add the specified record to the sort subtree 1`] = `
Object {
  "items": Array [],
  "sort": Object {
    "name": "asc",
    "size": "asc",
  },
}
`;

exports[`cases setField should update the specified record from the sort subtree if it exist 1`] = `
Object {
  "items": Array [],
  "sort": Object {
    "name": "desc",
    "size": "asc",
  },
}
`;

exports[`cases unsetField should remove the specified record from the sort subtree 1`] = `
Object {
  "items": Array [],
  "sort": Object {
    "name": "asc",
  },
}
`;

exports[`selectors getField should return the sort status of the key passed as param 1`] = `"desc"`;

exports[`selectors getFields should return all the sorts as an array 1`] = `
Array [
  Object {
    "size": "asc",
  },
  Object {
    "name": "asc",
  },
]
`;

exports[`sortingFactory should return default sort state 1`] = `Object {}`;

exports[`withSorting should be able to initialize the sort state  1`] = `
Object {
  "items": Array [],
  "sort": Object {
    "name": "asc",
  },
}
`;

exports[`withSorting should enhance the state with sort state 1`] = `
Object {
  "items": Array [],
  "sort": Object {},
}
`;
